.github/workflows/test.yml

```diff
@@ -0,0 +1,35 @@
+name: test
+on: [push, pull_request]
+
+jobs:
+  e2e-test:
+    runs-on: ubuntu-22.04
+    env:
+      SGX_MODE: SW
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
+      - uses: dtolnay/rust-toolchain@nightly
+      - uses: datachainlab/rust-cache@allow_registry_src_caching
+        with:
+          workspaces: |
+            ./lcp
+            ./enclave
+          cache-directories: |
+            ~/.cargo/registry/src/**/librocksdb-sys-*
+      - name: Install SGX SDK
+        run: |
+          curl -LO https://download.01.org/intel-sgx/sgx-linux/2.19/distro/ubuntu22.04-server/sgx_linux_x64_sdk_2.19.100.3.bin
+          chmod +x ./sgx_linux_x64_sdk_2.19.100.3.bin
+          echo -e 'no\n/opt' | ./sgx_linux_x64_sdk_2.19.100.3.bin
+      - name: Run E2E test
+        run: |
+          source /opt/sgxsdk/environment
+          make prepare-contracts
+          make build-images
+          make USE_UPGRADE_TEST=no e2e-test
```

Comment:
@siburu Is this correct? I think we should set `USE_UPGRADE_TEST` to `yes` in CI.